### PR TITLE
improve presentation of matching results and related

### PR DESCRIPTION
**Pull request by lifenjoiner**
https://github.com/stefankueng/grepWin/issues/83, https://github.com/stefankueng/grepWin/issues/117 and more. See the descriptions of each commit.

### CODE REVIEW

**Comment by stefankueng** [comment]
why do you call it "move"? Isn't this the column of the line?

File: src/SearchDlg.cpp [hunk 1]
```diff
@@ -2423,6 +2443,93 @@ bool CSearchDlg::PreTranslateMessage(MSG* pMsg)
...
+            return CDRF_NOTIFYSUBITEMDRAW;
+        case  CDDS_ITEMPREPAINT | CDDS_SUBITEM:
+            return CDRF_NOTIFYPOSTPAINT | CDRF_NEWFONT;
+        case CDDS_ITEMPOSTPAINT | CDDS_SUBITEM: // use the theme color
+        {
+            if (IsDlgButtonChecked(*this, IDC_RESULTFILES) != BST_CHECKED && lpLVCD->iSubItem == 3)
+            {
+                HDC     hdc             = lpLVCD->nmcd.hdc;
+                RECT    rc              = lpLVCD->nmcd.rc;  // lpLVCD->rcText does not work
+                if (rc.top == 0)
```
Comment (stefankueng) [inline src/SearchDlg.cpp:2462]: rc does not work either.
The only way that works reliably is to use LVM_GETSUBITEMRECT to get the rect.
https://learn.microsoft.com/en-us/windows/win32/controls/lvm-getsubitemrect

File: src/SearchDlg.cpp [hunk 2]
```diff
@@ -2423,6 +2443,93 @@ bool CSearchDlg::PreTranslateMessage(MSG* pMsg)
...
+    {
+        case CDDS_PREPAINT:                      // theme hooks this stage
+            return CDRF_NOTIFYITEMDRAW;
+        case CDDS_ITEMPREPAINT:
+            return CDRF_NOTIFYSUBITEMDRAW;
+        case  CDDS_ITEMPREPAINT | CDDS_SUBITEM:
+            return CDRF_NOTIFYPOSTPAINT | CDRF_NEWFONT;
+        case CDDS_ITEMPOSTPAINT | CDDS_SUBITEM: // use the theme color
+        {
+            if (IsDlgButtonChecked(*this, IDC_RESULTFILES) != BST_CHECKED && lpLVCD->iSubItem == 3)
```
Comment (stefankueng) [inline src/SearchDlg.cpp]: small performance issue: do the check `lpLVCD->iSubItem == 3` before `IsDlgButtonChecked(*this, IDC_RESULTFILES) != BST_CHECKED`

**Comment by lifenjoiner** [comment]
> why do you call it "move"? Isn't this the column of the line?

Many people take `\t` as indentations. And how many columns a `\t` counts is decided by the editor they use.
For example, :
```c
	int i = 0;
```
Here we can only say `int` starts from the 2nd char. In an editor, if the `\t` width is defined 4, then it is the 5th column; if 8, it is the 9th column.

BTW:
`Scintilla` based editors, `Sublime Text`, `UltraEdit` and `Vim`, that I checked, works in this way.
MS `Notepad` is the only one that I know of counting `\t` as 1 column but display it with 8 columns.

File: src/SearchDlg.cpp (hunk 2 above)
Comment (lifenjoiner) [reply src/SearchDlg.cpp]: Yeah, will be resolved.

File: src/SearchDlg.cpp (hunk 1 above)
Comment (lifenjoiner) [reply src/SearchDlg.cpp:2462]: I tested it on Win10 and Win7. It works.
`rc.top == 0` happens when we hover on the sub-item. This doesn't need `AlphaBlend`, as the display is not erased.

Could you tell me when it does not work? Then I can test for that.

**Comment by lifenjoiner** [comment]
I called it "column" at first.
"Position" has been used as `posMatch` in `SearchFile`.
"Offset", in many scenarios, is defined beginning from the start of the file.
They both give me the impression that from the start of the file.
Finally, "move" came into my mind, represents the times of caret moves.

File: src/SearchDlg.cpp (hunk 1 above)
Comment (stefankueng) [reply src/SearchDlg.cpp:2462]: using Win11, the nmcd.rc when hovering covers not just the subitem but all columns on the right as well. Which then produces the mark rect to be drawn on those columns.

**Comment by stefankueng** [comment]
while "caret move" would be better than just "move", I don't think either is good. It's not a common term for what it is used for. I myself had no idea what this "move" should be before checking the source code.
So this has to be changed to "column" or "line offset", anything but "move"...

**Comment by lifenjoiner** [comment]
None is perfect. I don't have a better one. So, it will be changed back to "column", and with explanation "line offset".

File: src/SearchDlg.cpp (hunk 1 above)
Comment (lifenjoiner) [reply src/SearchDlg.cpp:2462]: I'm not sure if I got you here. On my side, it also has no issue on Win11.
No mater `rc.top` equals 0 or not, it has no issue. See the screen-casts:

1. Win11-hover-no-item-highlight
   Win11-hover-no-item-highlight

2. Win10-hover
   Win10-hover

If I missed anything, tell me.

Comment (stefankueng) [reply src/SearchDlg.cpp:2462]: grepwin
as you can see, the yellow rect is drawn over the path column.

Comment (lifenjoiner) [reply src/SearchDlg.cpp:2462]: Oh... Yes, and thought that still indicates where to find the match.
Fixed by [f2c00ea](https://github.com/stefankueng/grepWin/commit/f2c00eac8e4bc445306fe5e79418aa110a19d3af).

**Pull request merged**